### PR TITLE
chore(Badge/MenuToggle): updated implementation of badge toggle

### LINKED ISSUE #6407: [Badge] - Remove badge with toggle variant
Follow up to https://github.com/patternfly/patternfly-design-kit/issues/716

We currently have a Variant on the Badge component used in Breadcrumbs that acts as a menu toggle. We can instead use a plain menu toggle that pulls in the Badge next to the caret.

### PR DESCRIPTION
**Pull request by thatblindgeye**
Closes #6407

### CODE REVIEW

**Review by mcoker — APPROVED** [review]
LGTM though I suspect we'll need to address the styling. Doesn't have to be part of this PR though. I'm curious what @andrew-ronaldson and @lboehling think.

Some things worth noting
* The menu toggle has a small variation that may work better?
* We should make sure the height of the breadcrumb doesn't change depending on whether there is a menu toggle in it or not (it changes currently)
* The toggle icon font-size is larger than the breadcrumb and badge text

**Comment by andrew-ronaldson** [comment]
I agree about that the height shouldn't change in breadcrumbs.
I noticed the spacing on this seems a bit large between the badge and the icon. Also I think there should be a `spacer-sm` betweent the > dividers so the hover state of menu toggle isn't touching the dividers

Screenshot 2024-03-18 at 12 07 09 PM

**Comment by mcoker** [comment]
@andrew-ronaldson we have a breadcrumb issue to look at some things and I added a note about the spacing around the dividers https://github.com/patternfly/patternfly/issues/6097

For the menu toggle, maybe we can look at this spacing. This is effectively a plain text menu toggle (@thatblindgeye I think we should use that variant instead of the plain - wdyt?) and here's how that spacing is created. There is a space between the "text" and the toggle icon, then there is a bunch of space around the toggle icon to effectively maintain some sort of minimum width. For this toggle, how would you expect the spacing to look like? IMO this should be a follow-up, and we should at least review the way we're creating that spacing, I think the way it is will not create the spacing in figma.

File: src/patternfly/components/Breadcrumb/examples/Breadcrumb.md [hunk 1]
```diff
@@ -140,7 +140,17 @@ import './Breadcrumb.css'
     {{#> breadcrumb-item}}
       {{> breadcrumb-item-divider}}
       {{#> breadcrumb-dropdown}}
-        {{> dropdown dropdown--id="dropdown-badge-toggle" dropdown--template--Breadcrumb="true" dropdown--IsExpanded="true"}}
+        {{#> menu-toggle menu-toggle--IsPlain=true}}
```
[the remark this answers is not in the document]
Comment (thatblindgeye) [reply src/patternfly/components/Breadcrumb/examples/Breadcrumb.md:141]: Per discussion off GH, we won't need to add a Menu to this example.

**Review by andrew-ronaldson — APPROVED** [review]

**Comment by thatblindgeye** [comment]
Opened https://github.com/patternfly/patternfly/issues/6463 as a followup

**Pull request merged**